### PR TITLE
Addsetfield

### LINKED ISSUE #23184: setfield

### PR DESCRIPTION
**Pull request by assem1390**
# PR Description

## Related Issue

Close #23184

## Checklist

- [ ] Did you add a function? yes
- [ ] Did you add the tests? yes
- [ ] Did you follow the steps we provided? to the best of my understanding

### CODE REVIEW

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]
Hey @assem1390 , can you please fix the failing tests ? Thanks

**Review by assem1390 — COMMENTED** [review]
request has changed

**Comment by assem1390** [comment]
Hi @Aarsh2001 I have fixed the failing test, but now new failing test that previously passed appears

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -407,6 +407,28 @@ def tofile(self, fid, /, sep="", format_="%s"):
...
             f.write(string)
 
+    def setfield(self, val, dtype, offset=0):
+        new_dtype = np_frontend.to_ivy_dtype(dtype)
+
+        # Check if the new dtype is valid for the given valuee.
+        try:
+            cast_val = ivy.cast(val, new_dtype)
+        except Exception as e:
+            raise ValueError("Cannot cast value to the given dtype.") from e
```
Comment (Sarvesh-Kesharwani) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:417]: Please remove this try catch block to allow the exception to propagate down to the inner call for improved debugging.

**Review by Sarvesh-Kesharwani — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**